src/command_controller.rs

```diff
@@ -248,34 +326,45 @@ where
 /// * `resource` - The resource where to set the OwnerReference
 /// * `owner_reference` - The OwnerReference to add
 ///
-async fn patch_owner_reference<T>(
+async fn patch_owner_reference<C>(
```

Comment:
I'm in a rush but I think this can be much simpler by using json_patch.
If you like I can make a suggestion next week.